csharp/src/Apache.Arrow/Ipc/ArrowStreamReader.cs

```diff
@@ -28,12 +28,18 @@ public class ArrowStreamReader : IArrowReader, IDisposable
     {
         public Schema Schema { get; protected set; }
         public Stream BaseStream { get; }
+        private readonly bool _leaveOpen;
 
         protected ArrayPool<byte> Buffers { get; }
 
-        public ArrowStreamReader(Stream stream)
+        public ArrowStreamReader(Stream stream) : this(stream, leaveOpen: false)
```

Comment:
side question: would it be beneficial for the `ArrowFileReader/Writer` to have this constructor parameter as well?

Reply:
Maybe?  In what situation does someone do `new ArrowFileReader(stream)` vs `new ArrowStreamReader(stream)`?

Reply:
Arrow has 2 formats:

1. An Arrow "stream" - https://arrow.apache.org/docs/format/IPC.html#streaming-format
2. An Arrow "file" - https://arrow.apache.org/docs/format/IPC.html#file-format

In some cases they are working with a "stream" format. In some cases they are working with a "file" format.

Reply:
Aha, the "stream" vs "file" isn't about whether working with a .NET `Stream` or with a path/`FileStream`, rather it's about the actual data format, regardless of where/how it's stored.  In that case, yes, it should also have these `leaveOpen` constructors.